app/models/comment.rb

```diff
@@ -1,4 +1,15 @@
 class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :post
+
+  after_create :increment_post_comments_counter
+  after_destroy :decrement_post_comments_counter
+
+  def increment_post_comments_counter
+    post.increment!(:comments_counter)
+  end
```

Comment:
[OPTIONAL] Great work for using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html). There's a better way with lesser line of code.
- You can use `after_save` to call the `#update_comment_counter` method every time a new comment has been saved to database, then make the `update_comment_counter` method private.
-  It should look something similar to this. :point_down: 

```
 # app/models/post.rb
  after_save :update_comment_counter

  private

  def update_comment_counter
    post.increment!(:comments_counter)
  end
```